backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/attachment/AttachmentRestService.java

```diff
@@ -0,0 +1,172 @@
+			@NonNull final AttachmentEntryService attachmentEntryService,
+			@NonNull final ExternalReferenceRestControllerService externalReferenceService,
+			@NonNull final ExternalReferenceTypes externalReferenceTypes)
+	{
+		this.attachmentEntryService = attachmentEntryService;
+		this.externalReferenceService = externalReferenceService;
+		this.externalReferenceTypes = externalReferenceTypes;
+	}
+
+	@NonNull
+	public JsonAttachmentResponse createAttachment(@NonNull final JsonAttachmentRequest jsonAttachmentRequest)
+	{
+		return trxManager.callInNewTrx(() -> createAttachmentWithTrx(jsonAttachmentRequest));
+	}
+
+	@NonNull
+	private JsonAttachmentResponse createAttachmentWithTrx(@NonNull final JsonAttachmentRequest jsonAttachmentRequest)
+	{
+		final JsonAttachment attachment = jsonAttachmentRequest.getAttachment();
+
+		final AttachmentTags attachmentTags = extractAttachmentTags(attachment.getTags());
+
+		final byte[] data = Base64.getDecoder().decode(attachment.getData().getBytes());
+
+		final String contentType = attachment.getMimeType() != null
+				? MimeType.getExtensionByType(attachment.getMimeType())
+				: MimeType.getMimeType(attachment.getFileName());
+
+		final AttachmentEntryCreateRequest request = AttachmentEntryCreateRequest.builder()
+				.type(AttachmentEntry.Type.Data)
+				.filename(attachment.getFileName())
+				.contentType(contentType)
+				.tags(attachmentTags)
+				.data(data)
+				.build();
+
+		final List<TableRecordReference> references = jsonAttachmentRequest.getTargets()
+				.stream()
+				.map(this::extractTableRecordReference)
+				.collect(ImmutableList.toImmutableList());
+
+		final AttachmentEntry entry = attachmentEntryService.createNewAttachment(references, request);
+
+		return JsonAttachmentResponse.builder()
+				.target(jsonAttachmentRequest.getTargets())
+				.attachmentId(String.valueOf(entry.getId().getRepoId()))
+				.filename(entry.getFilename())
+				.mimeType(entry.getMimeType())
+				.build();
+	}
+
+	@NonNull
+	private TableRecordReference extractTableRecordReference(@NonNull final JsonExternalReferenceTarget target)
+	{
+		final IExternalReferenceType targetType = extractTargetType(target.getExternalReferenceType());
+		final MetasfreshId metasfreshId = extractTargetIdentifier(target.getExternalReferenceIdentifier(), targetType);
+
+		return TableRecordReference.of(targetType.getTableName(), metasfreshId.getValue());
+	}
+
+	private MetasfreshId extractTargetIdentifier(
+			@NonNull final String externalTargetIdentifier,
+			@NonNull final IExternalReferenceType externalReferenceType)
+	{
+
+		final ExternalIdentifier targetIdentifier = ExternalIdentifier.of(externalTargetIdentifier);
+
+		switch (targetIdentifier.getType())
+		{
+			case METASFRESH_ID:
+				return targetIdentifier.asMetasfreshId();
+			case EXTERNAL_REFERENCE:
+				return externalReferenceService
+						.getJsonMetasfreshIdFromExternalReference((OrgId)null, targetIdentifier, externalReferenceType)
```

Comment:
I'm not sure this is ok as the `OrgId` is not present. This is the proposed solution.